ksql-api/src/main/java/io/confluent/ksql/api/server/BlockingQueryPublisher.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.server;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import io.confluent.ksql.GenericRow;
+import io.confluent.ksql.api.spi.QueryPublisher;
+import io.confluent.ksql.query.RowConsumer;
+import io.vertx.core.Context;
+import io.vertx.core.WorkerExecutor;
+import java.util.Objects;
+import java.util.OptionalInt;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A query publisher that uses an internal blocking queue to store rows for delivery. It's currently
+ * necessary to use a blocking queue as Kafka Streams delivers message in a synchronous fashion with
+ * no back pressure. If the queue was not blocking then if the subscriber was slow the messages
+ * could build up on the queue eventually resulting in out of memory. The only mechanism we have to
+ * slow streams down is to block the thread. Kafka Streams uses dedicated streams per topology so
+ * this won't prevent the thread from doing useful work elsewhere but it does mean we can't have too
+ * many push queries in the server at any one time as we can end up with a lot of threads. Ideally
+ * Kafka Streams would use a non-blocking reactive model with back-pressure, e.g. using reactive
+ * streams. That way a small number of threads would be required to service all topologies and we
+ * wouldn't need to block.
+ */
```

Comment:
Try to avoid such 'I wish Streams did it this way' style comments in the code.   Such points are for discussions, not comments in code.

Reply:
+1 one this one